From 62b19cddc66a4dc235c2a716562d844dceba809b Mon Sep 17 00:00:00 2001
From: huntxu <mhuntxu@gmail.com>
Date: Thu, 7 May 2015 12:44:34 +0800
Subject: [PATCH 2/3] vpn: update cmdline options and config for ipsec

This commit is based on the solution comes from the link below:
    http://www.gossamer-threads.com/lists/openstack/operators/44334

* ipsec pluto
  - Remove --use-netkey because it is the default option
* ipsec addconn
  - Remove --defaultroutenexthop because it is obsolete
* ipsec.conf.template
  - Remove obsolete options

Signed-off-by: huntxu <mhuntxu@gmail.com>
---
 neutron/services/vpn/device_drivers/ipsec.py                       | 3 ---
 .../vpn/device_drivers/template/openswan/ipsec.conf.template       | 7 ++-----
 2 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/neutron/services/vpn/device_drivers/ipsec.py b/neutron/services/vpn/device_drivers/ipsec.py
index 9330a4a..36d2ca5 100644
--- a/neutron/services/vpn/device_drivers/ipsec.py
+++ b/neutron/services/vpn/device_drivers/ipsec.py
@@ -378,7 +378,6 @@ class OpenSwanProcess(BaseSwanProcess):
                        'pluto',
                        '--ctlbase', self.pid_path,
                        '--ipsecdir', self.etc_dir,
-                       '--use-netkey',
                        '--uniqueids',
                        '--nat_traversal',
                        '--secretsfile', self.secrets_file,
@@ -386,11 +385,9 @@ class OpenSwanProcess(BaseSwanProcess):
                        ])
         #add connections
         for ipsec_site_conn in self.vpnservice['ipsec_site_connections']:
-            nexthop = self._get_nexthop(ipsec_site_conn['peer_address'])
             self._execute([self.binary,
                            'addconn',
                            '--ctlbase', '%s.ctl' % self.pid_path,
-                           '--defaultroutenexthop', nexthop,
                            '--config', self.config_file,
                            ipsec_site_conn['id']
                            ])
diff --git a/neutron/services/vpn/device_drivers/template/openswan/ipsec.conf.template b/neutron/services/vpn/device_drivers/template/openswan/ipsec.conf.template
index 546e27e..3da5cd8 100644
--- a/neutron/services/vpn/device_drivers/template/openswan/ipsec.conf.template
+++ b/neutron/services/vpn/device_drivers/template/openswan/ipsec.conf.template
@@ -1,10 +1,9 @@
 # Configuration for {{vpnservice.name}}
 config setup
-    nat_traversal=yes
     listen={{vpnservice.external_ip}}
 conn %default
     ikelifetime=480m
-    keylife=60m
+    salifetime=60m
     keyingtries=%forever
 {% for ipsec_site_connection in vpnservice.ipsec_site_connections if ipsec_site_connection.admin_state_up
 %}conn {{ipsec_site_connection.id}}
@@ -16,7 +15,6 @@ conn %default
     # [subnet]
     leftsubnet={{vpnservice.subnet.cidr}}
     # leftsubnet=networkA/netmaskA, networkB/netmaskB (IKEv2 only)
-    leftnexthop=%defaultroute
     ######################
     # ipsec_site_connections
     ######################
@@ -27,7 +25,6 @@ conn %default
     # [peer_cidrs]
     rightsubnets={ {{ipsec_site_connection['peer_cidrs']|join(' ')}} }
     # rightsubnet=networkA/netmaskA, networkB/netmaskB (IKEv2 only)
-    rightnexthop=%defaultroute
     # [mtu]
     # Note It looks like not supported in the strongswan driver
     # ignore it now
@@ -59,6 +56,6 @@ conn %default
     # [encapsulation_mode]
     type={{ipsec_site_connection.ipsecpolicy.encapsulation_mode}}
     # [lifetime_value]
-    lifetime={{ipsec_site_connection.ipsecpolicy.lifetime_value}}s
+    salifetime={{ipsec_site_connection.ipsecpolicy.lifetime_value}}s
     # lifebytes=100000 if lifetime_units=kilobytes (IKEv2 only)
 {% endfor %}
-- 
2.4.0

